packages/snap/src/keyring.ts

```diff
@@ -218,11 +223,17 @@ export class AccountAbstractionKeyring implements Keyring {
     options: Record<string, Json> = {},
   ): Promise<KeyringAccount> {
     if (!options.privateKey) {
-      throwError(`[Snap] Private Key is required`);
+      if (!options.saltIndex) {
+        throwError(`[Snap] Private Key or Salt Index is required`);
+      }
     }
```

Comment:
if(!options.privateKey || !options.saltIndex) {
        throwError(`[Snap] Private Key or Salt Index is required`);
}